File: frontend/shared/types/customer.ts
export interface Customer {
  id: number
  name: string
  phone?: string | null
  lingkar_badan?: number | null
  lingkar_pinggang?: number | null
  lingkar_panggul?: number | null
  panjang_bahu?: number | null
  panjang_tgn?: number | null
  panjang_baju?: number | null
  panjang_rok?: number | null
  createdAt?: string
  updatedAt?: string | null
  total_orders: number
}

export interface CustomerBrief {
  id: number
  name: string
  phone?: string | null
}

export interface CustomerCreate {
  name: string
  phone?: string | null
  lingkar_badan?: number | null
  lingkar_pinggang?: number | null
  lingkar_panggul?: number | null
  panjang_bahu?: number | null
  panjang_tgn?: number | null
  panjang_baju?: number | null
  panjang_rok?: number | null
}

export interface CustomerUpdate {
  name?: string | null
  phone?: string | null
  lingkar_badan?: number | null
  lingkar_pinggang?: number | null
  lingkar_panggul?: number | null
  panjang_bahu?: number | null
  panjang_tgn?: number | null
  panjang_baju?: number | null
  panjang_rok?: number | null
}

export interface CustomerOrderItem {
  id: number
  receipt_number: string
  total_price: number
  paid_amount: number
  payment_status: string
  status: string
  created_at: string
}

export interface CustomerDetail {
  customer: Customer
  orders: CustomerOrderItem[]
  total_bon: number
}

File: backend/app/schemas/customer.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class CustomerBase(BaseModel):
    name: str
    phone: Optional[str] = None
    lingkar_badan: Optional[float] = None
    lingkar_pinggang: Optional[float] = None
    lingkar_panggul: Optional[float] = None
    panjang_bahu: Optional[float] = None
    panjang_tgn: Optional[float] = None
    panjang_baju: Optional[float] = None
    panjang_rok: Optional[float] = None


class CustomerUpdate(BaseModel):
    name: Optional[str] = None
    phone: Optional[str] = None
    lingkar_badan: Optional[float] = None
    lingkar_pinggang: Optional[float] = None
    lingkar_panggul: Optional[float] = None
    panjang_bahu: Optional[float] = None
    panjang_tgn: Optional[float] = None
    panjang_baju: Optional[float] = None
    panjang_rok: Optional[float] = None


class CustomerBrief(BaseModel):
    id: int
    name: str
    phone: Optional[str] = None

    class Config:
        from_attributes = True


class Customer(CustomerBase):
    id: int
    createdAt: datetime
    updatedAt: Optional[datetime] = None
    total_orders: int = 0

    class Config:
        from_attributes = True


class CustomerOrderItem(BaseModel):
    id: int
    receipt_number: str
    total_price: float
    dp_amount: float
    payment_status: str
    status: str
    created_at: datetime


class CustomerDetail(BaseModel):
    customer: Customer
    orders: List[CustomerOrderItem] = []
    total_bon: float = 0